app/components/ui-table.js

```diff
@@ -1,130 +1,12 @@
-  footerSummaryDefaultPagination : 'column four wide',
-  pageSizeWrapper                : 'ui column eight wide grid',
-  pageSizeSelectWrapper          : 'left aligned',
-  paginationWrapper              : 'ui column right floated',
-  paginationWrapperNumeric       : 'column four wide',
-  paginationWrapperDefault       : 'column four wide',
-  buttonDefault                  : 'ui basic button',
-  noDataCell                     : '',
-  collapseRow                    : 'collapse-row',
-  collapseAllRows                : 'collapse-all-rows',
-  expandRow                      : 'expand-row',
-  expandAllRows                  : 'expand-all-rows',
-  thead                          : '',
-  input                          : '',
-  clearFilterIcon                : 'remove circle icon',
-  clearAllFiltersIcon            : 'remove circle outline icon',
-  globalFilterDropdownWrapper    : 'ui row stackable grid'
-};
-
-const assign = Object.assign || assign;
+import Semantic from 'open-event-frontend/themes/semantic';
+import { computed } from '@ember/object';
 
 export default TableComponent.extend({
   layout,
 
-  _setupMessages: observer('customMessages', function() {
-    const customIcons = getWithDefault(this, 'customMessages', {});
-    let newMessages = {};
-    assign(newMessages, defaultMessages, customIcons);
-    set(this, 'messages', O.create(newMessages));
-  }),
-
-  _setupIcons() {
-    const customIcons = getWithDefault(this, 'customIcons', {});
-    let newIcons = {};
-    assign(newIcons, defaultIcons, customIcons);
-    set(this, 'icons', O.create(newIcons));
-  },
-
-  _setupClasses() {
-    const customClasses = getWithDefault(this, 'customClasses', {});
-    let newClasses = {};
-    assign(newClasses, defaultCssClasses, customClasses);
-    set(this, 'classes', O.create(newClasses));
-  },
-
-  simplePaginationTemplate: 'components/ui-table/simple-pagination',
-
-  numericPaginationTemplate: 'components/ui-table/numeric-pagination',
-
-  tableFooterTemplate: 'components/ui-table/table-footer',
-
-  componentFooterTemplate: 'components/ui-table/component-footer',
-
-  pageSizeTemplate: 'components/ui-table/page-size',
-
-  showColumnsDropdown: false,
-
-  useFilteringByColumns: false,
-
-  globalFilterTemplate: 'components/ui-table/global-filter',
-
-  columnsDropdownTemplate: 'components/ui-table/columns-dropdown',
-
-  headerSortingRowTemplate: 'components/ui-table/header-row-sorting',
-
-  headerSortingIconsTemplate: 'components/ui-table/header-sorting-icons',
-
-  headerFilteringRowTemplate: 'components/ui-table/header-row-filtering',
-
-  headerGroupedRowsTemplate: 'components/ui-table/header-rows-grouped',
-
-  rowTemplate: 'components/ui-table/row',
-
-  expandedRowTemplate: 'components/ui-table/expanded-row',
-
-  noDataShowTemplate: 'components/ui-table/no-data',
-
-  allColumnsHiddenTemplate: 'components/ui-table/all-columns-hidden'
-
+  themeInstance: computed(function() {
```

Comment:
I don't think this needs to be a computed property. `themeInstance: Semantic.create()` should do.